kyuubi-server/src/main/scala/org/apache/kyuubi/server/http/ThriftHttpServlet.scala

```diff
@@ -0,0 +1,539 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.http
+
+import java.io.IOException
+import java.security.PrivilegedExceptionAction
+import java.security.SecureRandom
+import javax.security.sasl.AuthenticationException
+import javax.servlet.ServletException
+import javax.servlet.http.Cookie
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import javax.ws.rs.core.NewCookie
+
+import scala.collection.mutable
+
+import org.apache.commons.codec.binary.Base64
+import org.apache.commons.codec.binary.StringUtils
+import org.apache.hadoop.hive.shims.HadoopShims.KerberosNameShim
+import org.apache.hadoop.hive.shims.ShimLoader
+import org.apache.hadoop.hive.shims.Utils
+import org.apache.hadoop.security.UserGroupInformation
+import org.apache.thrift.TProcessor
+import org.apache.thrift.protocol.TProtocolFactory
+import org.apache.thrift.server.TServlet
+import org.ietf.jgss.GSSContext
+import org.ietf.jgss.GSSCredential
+import org.ietf.jgss.GSSException
+import org.ietf.jgss.GSSManager
+import org.ietf.jgss.Oid
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.server.http.authentication.KerberosAuthenticationHandler
+import org.apache.kyuubi.server.http.util.{CookieSigner, HttpAuthUtils, SessionManager}
+import org.apache.kyuubi.service.authentication.{AuthenticationProviderFactory, KyuubiAuthenticationFactory}
+
+class ThriftHttpServlet(
+    processor: TProcessor,
+    protocolFactory: TProtocolFactory,
+    authFactory: KyuubiAuthenticationFactory,
+    conf: KyuubiConf)
+  extends TServlet(processor, protocolFactory) with Logging {
+  // Class members for cookie based authentication.
+  private var signer: CookieSigner = _
+  val AUTH_COOKIE = "hive.server2.auth"
+  private val RAN = new SecureRandom()
+  private var isCookieAuthEnabled: Boolean = false
+  private var cookieDomain: String = _
+  private var cookiePath: String = _
+  private var cookieMaxAge: Int = 0
+  private var isCookieSecure = false
+  private var isHttpOnlyCookie = false
+  private val HIVE_DELEGATION_TOKEN_HEADER = "X-Hive-Delegation-Token"
+  private val X_FORWARDED_FOR = "X-Forwarded-For"
```

Comment:
Please unify the name style of `HIVE_DELEGATION_TOKEN_HEADER` and `X_FORWARDED_FOR`